From 9cf407e9980e8459766a8df8a30cf6d6e79e1c54 Mon Sep 17 00:00:00 2001
From: Michael Lippautz <mlippautz@chromium.org>
Date: Tue, 30 Aug 2022 12:44:33 +0200
Subject: [PATCH] Fix includes of base/platform/wrappers.h

Follow up from https://crrev.com/c/3858226

- Remove include when not necessary.
- Choose platform/memory.h when appropriate.

Change-Id: Ic24ef89aefb754acccef896e6b59a7ba744ac2df
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3862504
Reviewed-by: Camillo Bruni <cbruni@chromium.org>
Commit-Queue: Adam Klein <adamk@chromium.org>
Auto-Submit: Michael Lippautz <mlippautz@chromium.org>
Reviewed-by: Adam Klein <adamk@chromium.org>
Reviewed-by: Igor Sheludko <ishell@chromium.org>
Cr-Commit-Position: refs/heads/main@{#83007}
---
 src/ast/ast.cc                                |  1 -
 src/base/emulated-virtual-address-subspace.cc |  1 -
 src/base/file-utils.cc                        |  1 -
 src/base/free_deleter.h                       |  2 +-
 src/base/hashmap.h                            |  2 +-
 src/base/memory.h                             |  1 -
 src/base/page-allocator.cc                    |  1 -
 src/base/platform/wrappers.h                  | 46 +++++++++----------
 src/base/virtual-address-space.cc             |  1 -
 src/builtins/builtins-trace.cc                |  1 -
 src/codegen/arm64/assembler-arm64.h           |  1 -
 src/codegen/arm64/macro-assembler-arm64.cc    |  1 -
 src/codegen/external-reference.cc             |  1 -
 src/codegen/optimized-compilation-info.cc     |  1 -
 src/codegen/x64/assembler-x64.cc              |  1 -
 .../backend/arm/instruction-selector-arm.cc   |  1 -
 .../arm64/instruction-selector-arm64.cc       |  1 -
 .../backend/ia32/instruction-selector-ia32.cc |  1 -
 src/compiler/backend/instruction-selector.cc  |  1 -
 .../loong64/instruction-selector-loong64.cc   |  1 -
 .../mips64/instruction-selector-mips64.cc     |  1 -
 .../backend/s390/instruction-selector-s390.cc |  1 -
 .../backend/x64/instruction-selector-x64.cc   |  1 -
 src/d8/d8.cc                                  |  1 +
 src/debug/wasm/gdb-server/transport.cc        |  2 -
 src/diagnostics/unwinding-info-win64.cc       |  1 -
 src/execution/arm/simulator-arm.cc            |  2 +-
 src/execution/arm64/simulator-arm64.h         |  1 -
 src/execution/frames.cc                       |  1 -
 src/execution/loong64/simulator-loong64.cc    |  2 +-
 src/execution/mips/simulator-mips.cc          |  2 +-
 src/execution/mips64/simulator-mips64.cc      |  2 +-
 src/execution/ppc/simulator-ppc.cc            |  2 +-
 src/execution/ppc/simulator-ppc.h             |  1 -
 src/execution/s390/simulator-s390.cc          |  2 +-
 src/heap/heap.cc                              |  2 +-
 src/heap/memory-chunk.cc                      |  1 -
 src/libplatform/tracing/trace-object.cc       |  1 -
 src/numbers/conversions-inl.h                 |  1 -
 src/numbers/conversions.cc                    |  1 -
 src/objects/backing-store.cc                  |  1 -
 src/objects/bigint.cc                         |  1 -
 src/objects/fixed-array-inl.h                 |  1 -
 src/objects/js-array-buffer.cc                |  1 -
 src/objects/shared-function-info-inl.h        |  1 -
 src/objects/value-serializer.cc               |  2 +-
 src/parsing/preparse-data.cc                  |  1 -
 src/parsing/scanner.cc                        |  1 -
 src/snapshot/embedded/embedded-file-writer.h  |  2 +-
 src/tracing/trace-event.h                     |  1 -
 src/utils/allocation.cc                       |  1 -
 src/utils/memcopy.h                           |  1 -
 src/wasm/baseline/arm/liftoff-assembler-arm.h |  1 -
 .../baseline/arm64/liftoff-assembler-arm64.h  |  1 -
 .../baseline/ia32/liftoff-assembler-ia32.h    |  1 -
 src/wasm/baseline/liftoff-assembler.cc        |  2 +-
 src/wasm/baseline/liftoff-compiler.cc         |  1 -
 .../loong64/liftoff-assembler-loong64.h       |  1 -
 .../baseline/mips/liftoff-assembler-mips.h    |  1 -
 .../mips64/liftoff-assembler-mips64.h         |  1 -
 src/wasm/baseline/ppc/liftoff-assembler-ppc.h |  1 -
 .../baseline/riscv/liftoff-assembler-riscv.h  |  1 -
 .../riscv/liftoff-assembler-riscv32.h         |  1 -
 .../riscv/liftoff-assembler-riscv64.h         |  1 -
 .../baseline/s390/liftoff-assembler-s390.h    |  1 -
 src/wasm/baseline/x64/liftoff-assembler-x64.h |  1 -
 src/wasm/c-api.cc                             |  1 -
 src/wasm/module-decoder-impl.h                |  1 +
 src/wasm/wasm-code-manager.cc                 |  2 +-
 src/wasm/wasm-engine.cc                       |  1 -
 src/web-snapshot/web-snapshot.cc              |  1 -
 src/zone/accounting-allocator.cc              |  1 -
 src/zone/zone-list-inl.h                      |  1 -
 73 files changed, 36 insertions(+), 96 deletions(-)

diff --git a/src/ast/ast.cc b/src/ast/ast.cc
index 97db02225f8..34a26d9c269 100644
--- a/src/ast/ast.cc
+++ b/src/ast/ast.cc
@@ -12,7 +12,6 @@
 #include "src/base/hashmap.h"
 #include "src/base/logging.h"
 #include "src/base/numbers/double.h"
-#include "src/base/platform/wrappers.h"
 #include "src/builtins/builtins-constructor.h"
 #include "src/builtins/builtins.h"
 #include "src/common/assert-scope.h"
diff --git a/src/base/emulated-virtual-address-subspace.cc b/src/base/emulated-virtual-address-subspace.cc
index c121ca4ecd8..a61102008f7 100644
--- a/src/base/emulated-virtual-address-subspace.cc
+++ b/src/base/emulated-virtual-address-subspace.cc
@@ -6,7 +6,6 @@
 
 #include "src/base/bits.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 
 namespace v8 {
 namespace base {
diff --git a/src/base/file-utils.cc b/src/base/file-utils.cc
index 2fa729a1fa7..6e1c4921440 100644
--- a/src/base/file-utils.cc
+++ b/src/base/file-utils.cc
@@ -8,7 +8,6 @@
 #include <string.h>
 
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 
 namespace v8 {
 namespace base {
diff --git a/src/base/free_deleter.h b/src/base/free_deleter.h
index 4f4efcda17e..66d0f55c81e 100644
--- a/src/base/free_deleter.h
+++ b/src/base/free_deleter.h
@@ -12,7 +12,7 @@
 
 #include <memory>
 
-#include "src/base/platform/wrappers.h"
+#include "src/base/platform/memory.h"
 
 namespace v8 {
 namespace base {
diff --git a/src/base/hashmap.h b/src/base/hashmap.h
index 6648eff6b28..7130d623aa0 100644
--- a/src/base/hashmap.h
+++ b/src/base/hashmap.h
@@ -14,7 +14,7 @@
 #include "src/base/bits.h"
 #include "src/base/hashmap-entry.h"
 #include "src/base/logging.h"
-#include "src/base/platform/wrappers.h"
+#include "src/base/platform/memory.h"
 
 namespace v8 {
 namespace base {
diff --git a/src/base/memory.h b/src/base/memory.h
index db6a9ea1db3..e2676a82e77 100644
--- a/src/base/memory.h
+++ b/src/base/memory.h
@@ -6,7 +6,6 @@
 #define V8_BASE_MEMORY_H_
 
 #include "src/base/macros.h"
-#include "src/base/platform/wrappers.h"
 
 namespace v8 {
 namespace base {
diff --git a/src/base/page-allocator.cc b/src/base/page-allocator.cc
index d6da1abc343..7303c5ed589 100644
--- a/src/base/page-allocator.cc
+++ b/src/base/page-allocator.cc
@@ -5,7 +5,6 @@
 #include "src/base/page-allocator.h"
 
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 
 #if V8_OS_DARWIN
 #include <sys/mman.h>  // For MAP_JIT.
diff --git a/src/base/platform/wrappers.h b/src/base/platform/wrappers.h
index 70b913bd8a5..e47af5216e5 100644
--- a/src/base/platform/wrappers.h
+++ b/src/base/platform/wrappers.h
@@ -10,40 +10,36 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include "src/base/base-export.h"
-#include "src/base/platform/memory.h"
-
-#if defined(V8_OS_STARBOARD)
-#include "starboard/memory.h"
+#ifdef V8_OS_STARBOARD
 #include "starboard/string.h"
 #endif
 
-namespace v8 {
-namespace base {
-
-#if !defined(V8_OS_STARBOARD)
+namespace v8::base {
 
-// Common libstd implementations.
-// inline implementations are preferred here due to performance concerns.
-inline char* Strdup(const char* source) { return strdup(source); }
+inline char* Strdup(const char* source) {
+#if V8_OS_STARBOARD
+  return SbStringDuplicate(source);
+#else
+  return strdup(source);
+#endif
+}
 
 inline FILE* Fopen(const char* filename, const char* mode) {
+#if V8_OS_STARBOARD
+  return NULL;
+#else
   return fopen(filename, mode);
+#endif
 }
 
-inline int Fclose(FILE* stream) { return fclose(stream); }
-
-#else  // V8_OS_STARBOARD
-
-inline char* Strdup(const char* source) { return SbStringDuplicate(source); }
-
-inline FILE* Fopen(const char* filename, const char* mode) { return NULL; }
-
-inline int Fclose(FILE* stream) { return -1; }
-
-#endif  // V8_OS_STARBOARD
+inline int Fclose(FILE* stream) {
+#if V8_OS_STARBOARD
+  return -1;
+#else
+  return fclose(stream);
+#endif
+}
 
-}  // namespace base
-}  // namespace v8
+}  // namespace v8::base
 
 #endif  // V8_BASE_PLATFORM_WRAPPERS_H_
diff --git a/src/base/virtual-address-space.cc b/src/base/virtual-address-space.cc
index aea15abfcdb..490a112e630 100644
--- a/src/base/virtual-address-space.cc
+++ b/src/base/virtual-address-space.cc
@@ -7,7 +7,6 @@
 #include "include/v8-platform.h"
 #include "src/base/bits.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 
 namespace v8 {
 namespace base {
diff --git a/src/builtins/builtins-trace.cc b/src/builtins/builtins-trace.cc
index 15124da50d1..22941ddb0ff 100644
--- a/src/builtins/builtins-trace.cc
+++ b/src/builtins/builtins-trace.cc
@@ -13,7 +13,6 @@
 
 #if defined(V8_USE_PERFETTO)
 #include "protos/perfetto/trace/track_event/debug_annotation.pbzero.h"
-#include "src/base/platform/wrappers.h"
 #endif
 
 namespace v8 {
diff --git a/src/codegen/arm64/assembler-arm64.h b/src/codegen/arm64/assembler-arm64.h
index e079e87a205..368a2e5f42b 100644
--- a/src/codegen/arm64/assembler-arm64.h
+++ b/src/codegen/arm64/assembler-arm64.h
@@ -25,7 +25,6 @@
 #endif
 
 #if defined(V8_OS_WIN)
-#include "src/base/platform/wrappers.h"
 #include "src/diagnostics/unwinding-info-win64.h"
 #endif  // V8_OS_WIN
 
diff --git a/src/codegen/arm64/macro-assembler-arm64.cc b/src/codegen/arm64/macro-assembler-arm64.cc
index bab3bb22ff1..0733257e2db 100644
--- a/src/codegen/arm64/macro-assembler-arm64.cc
+++ b/src/codegen/arm64/macro-assembler-arm64.cc
@@ -31,7 +31,6 @@
 // Satisfy cpplint check, but don't include platform-specific header. It is
 // included recursively via macro-assembler.h.
 #if 0
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/arm64/macro-assembler-arm64.h"
 #endif
 
diff --git a/src/codegen/external-reference.cc b/src/codegen/external-reference.cc
index b1c52406c0a..57aa7e6ec38 100644
--- a/src/codegen/external-reference.cc
+++ b/src/codegen/external-reference.cc
@@ -41,7 +41,6 @@
 #endif  // V8_ENABLE_WEBASSEMBLY
 
 #ifdef V8_INTL_SUPPORT
-#include "src/base/platform/wrappers.h"
 #include "src/base/strings.h"
 #include "src/objects/intl-objects.h"
 #endif  // V8_INTL_SUPPORT
diff --git a/src/codegen/optimized-compilation-info.cc b/src/codegen/optimized-compilation-info.cc
index 603de092d2a..38f9887fcd2 100644
--- a/src/codegen/optimized-compilation-info.cc
+++ b/src/codegen/optimized-compilation-info.cc
@@ -5,7 +5,6 @@
 #include "src/codegen/optimized-compilation-info.h"
 
 #include "src/api/api.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/source-position.h"
 #include "src/debug/debug.h"
 #include "src/execution/isolate.h"
diff --git a/src/codegen/x64/assembler-x64.cc b/src/codegen/x64/assembler-x64.cc
index 796a74ce1c9..2bbf6fda821 100644
--- a/src/codegen/x64/assembler-x64.cc
+++ b/src/codegen/x64/assembler-x64.cc
@@ -17,7 +17,6 @@
 
 #include "src/base/bits.h"
 #include "src/base/cpu.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/macro-assembler.h"
 #include "src/deoptimizer/deoptimizer.h"
diff --git a/src/compiler/backend/arm/instruction-selector-arm.cc b/src/compiler/backend/arm/instruction-selector-arm.cc
index ba149e9e18d..578581877e3 100644
--- a/src/compiler/backend/arm/instruction-selector-arm.cc
+++ b/src/compiler/backend/arm/instruction-selector-arm.cc
@@ -5,7 +5,6 @@
 #include "src/base/bits.h"
 #include "src/base/enum-set.h"
 #include "src/base/iterator.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/machine-type.h"
 #include "src/compiler/backend/instruction-selector-impl.h"
 #include "src/compiler/node-matchers.h"
diff --git a/src/compiler/backend/arm64/instruction-selector-arm64.cc b/src/compiler/backend/arm64/instruction-selector-arm64.cc
index b8e46f6a65b..66c39e6c9de 100644
--- a/src/compiler/backend/arm64/instruction-selector-arm64.cc
+++ b/src/compiler/backend/arm64/instruction-selector-arm64.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 
 #include "src/base/bits.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/machine-type.h"
 #include "src/common/globals.h"
diff --git a/src/compiler/backend/ia32/instruction-selector-ia32.cc b/src/compiler/backend/ia32/instruction-selector-ia32.cc
index 0154f90f5c4..84904062cfb 100644
--- a/src/compiler/backend/ia32/instruction-selector-ia32.cc
+++ b/src/compiler/backend/ia32/instruction-selector-ia32.cc
@@ -14,7 +14,6 @@
 #include "src/base/iterator.h"
 #include "src/base/logging.h"
 #include "src/base/macros.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/cpu-features.h"
 #include "src/codegen/ia32/assembler-ia32.h"
 #include "src/codegen/ia32/register-ia32.h"
diff --git a/src/compiler/backend/instruction-selector.cc b/src/compiler/backend/instruction-selector.cc
index c27774c64e2..27cace8e426 100644
--- a/src/compiler/backend/instruction-selector.cc
+++ b/src/compiler/backend/instruction-selector.cc
@@ -7,7 +7,6 @@
 #include <limits>
 
 #include "src/base/iterator.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/interface-descriptors-inl.h"
 #include "src/codegen/tick-counter.h"
diff --git a/src/compiler/backend/loong64/instruction-selector-loong64.cc b/src/compiler/backend/loong64/instruction-selector-loong64.cc
index 7ea977c18a8..2498cdcb072 100644
--- a/src/compiler/backend/loong64/instruction-selector-loong64.cc
+++ b/src/compiler/backend/loong64/instruction-selector-loong64.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 
 #include "src/base/bits.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/machine-type.h"
 #include "src/compiler/backend/instruction-selector-impl.h"
 #include "src/compiler/node-matchers.h"
diff --git a/src/compiler/backend/mips64/instruction-selector-mips64.cc b/src/compiler/backend/mips64/instruction-selector-mips64.cc
index f384591574c..fc6972fa99a 100644
--- a/src/compiler/backend/mips64/instruction-selector-mips64.cc
+++ b/src/compiler/backend/mips64/instruction-selector-mips64.cc
@@ -3,7 +3,6 @@
 // found in the LICENSE file.
 
 #include "src/base/bits.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/machine-type.h"
 #include "src/compiler/backend/instruction-selector-impl.h"
 #include "src/compiler/node-matchers.h"
diff --git a/src/compiler/backend/s390/instruction-selector-s390.cc b/src/compiler/backend/s390/instruction-selector-s390.cc
index fee9fa98be3..fc49a1376e5 100644
--- a/src/compiler/backend/s390/instruction-selector-s390.cc
+++ b/src/compiler/backend/s390/instruction-selector-s390.cc
@@ -2,7 +2,6 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "src/base/platform/wrappers.h"
 #include "src/compiler/backend/instruction-selector-impl.h"
 #include "src/compiler/node-matchers.h"
 #include "src/compiler/node-properties.h"
diff --git a/src/compiler/backend/x64/instruction-selector-x64.cc b/src/compiler/backend/x64/instruction-selector-x64.cc
index 0558783760e..09d005bce99 100644
--- a/src/compiler/backend/x64/instruction-selector-x64.cc
+++ b/src/compiler/backend/x64/instruction-selector-x64.cc
@@ -7,7 +7,6 @@
 #include "src/base/iterator.h"
 #include "src/base/logging.h"
 #include "src/base/overflowing-math.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/cpu-features.h"
 #include "src/codegen/machine-type.h"
 #include "src/compiler/backend/instruction-codes.h"
diff --git a/src/d8/d8.cc b/src/d8/d8.cc
index 3816d1ac99e..b788a8a749b 100644
--- a/src/d8/d8.cc
+++ b/src/d8/d8.cc
@@ -34,6 +34,7 @@
 #include "src/api/api-inl.h"
 #include "src/base/cpu.h"
 #include "src/base/logging.h"
+#include "src/base/platform/memory.h"
 #include "src/base/platform/platform.h"
 #include "src/base/platform/time.h"
 #include "src/base/platform/wrappers.h"
diff --git a/src/debug/wasm/gdb-server/transport.cc b/src/debug/wasm/gdb-server/transport.cc
index 9806a72a63c..500ff71e88c 100644
--- a/src/debug/wasm/gdb-server/transport.cc
+++ b/src/debug/wasm/gdb-server/transport.cc
@@ -6,8 +6,6 @@
 
 #include <fcntl.h>
 
-#include "src/base/platform/wrappers.h"
-
 #ifndef SD_BOTH
 #define SD_BOTH 2
 #endif
diff --git a/src/diagnostics/unwinding-info-win64.cc b/src/diagnostics/unwinding-info-win64.cc
index 4a70a284785..37f02cbf1d2 100644
--- a/src/diagnostics/unwinding-info-win64.cc
+++ b/src/diagnostics/unwinding-info-win64.cc
@@ -10,7 +10,6 @@
 #if defined(V8_OS_WIN_X64)
 #include "src/codegen/x64/assembler-x64.h"
 #elif defined(V8_OS_WIN_ARM64)
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/arm64/assembler-arm64-inl.h"
 #include "src/codegen/arm64/macro-assembler-arm64-inl.h"
 #else
diff --git a/src/execution/arm/simulator-arm.cc b/src/execution/arm/simulator-arm.cc
index ab506cd749e..ee331f606a6 100644
--- a/src/execution/arm/simulator-arm.cc
+++ b/src/execution/arm/simulator-arm.cc
@@ -17,8 +17,8 @@
 #include "src/base/lazy-instance.h"
 #include "src/base/memory.h"
 #include "src/base/overflowing-math.h"
+#include "src/base/platform/memory.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/vector.h"
 #include "src/codegen/arm/constants-arm.h"
 #include "src/codegen/assembler-inl.h"
diff --git a/src/execution/arm64/simulator-arm64.h b/src/execution/arm64/simulator-arm64.h
index 44e6abf77af..0094619a48b 100644
--- a/src/execution/arm64/simulator-arm64.h
+++ b/src/execution/arm64/simulator-arm64.h
@@ -15,7 +15,6 @@
 #include <vector>
 
 #include "src/base/compiler-specific.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/arm64/assembler-arm64.h"
 #include "src/codegen/arm64/decoder-arm64.h"
 #include "src/codegen/assembler.h"
diff --git a/src/execution/frames.cc b/src/execution/frames.cc
index 4fbdf6a711a..6baaf4bd576 100644
--- a/src/execution/frames.cc
+++ b/src/execution/frames.cc
@@ -9,7 +9,6 @@
 #include <sstream>
 
 #include "src/base/bits.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/interface-descriptors.h"
 #include "src/codegen/macro-assembler.h"
 #include "src/codegen/maglev-safepoint-table.h"
diff --git a/src/execution/loong64/simulator-loong64.cc b/src/execution/loong64/simulator-loong64.cc
index 4fb2fdb2ce5..1e8702d5833 100644
--- a/src/execution/loong64/simulator-loong64.cc
+++ b/src/execution/loong64/simulator-loong64.cc
@@ -14,8 +14,8 @@
 #include <cmath>
 
 #include "src/base/bits.h"
+#include "src/base/platform/memory.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/strings.h"
 #include "src/base/vector.h"
 #include "src/codegen/assembler-inl.h"
diff --git a/src/execution/mips/simulator-mips.cc b/src/execution/mips/simulator-mips.cc
index 690d375f682..bcf2bb8f880 100644
--- a/src/execution/mips/simulator-mips.cc
+++ b/src/execution/mips/simulator-mips.cc
@@ -15,8 +15,8 @@
 
 #include "src/base/bits.h"
 #include "src/base/lazy-instance.h"
+#include "src/base/platform/memory.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/vector.h"
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/macro-assembler.h"
diff --git a/src/execution/mips64/simulator-mips64.cc b/src/execution/mips64/simulator-mips64.cc
index 7ec93fe7669..a0e8d22156b 100644
--- a/src/execution/mips64/simulator-mips64.cc
+++ b/src/execution/mips64/simulator-mips64.cc
@@ -14,8 +14,8 @@
 #include <cmath>
 
 #include "src/base/bits.h"
+#include "src/base/platform/memory.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/strings.h"
 #include "src/base/vector.h"
 #include "src/codegen/assembler-inl.h"
diff --git a/src/execution/ppc/simulator-ppc.cc b/src/execution/ppc/simulator-ppc.cc
index 749a4bd56c2..d3d260b1c14 100644
--- a/src/execution/ppc/simulator-ppc.cc
+++ b/src/execution/ppc/simulator-ppc.cc
@@ -14,8 +14,8 @@
 #include "src/base/bits.h"
 #include "src/base/lazy-instance.h"
 #include "src/base/overflowing-math.h"
+#include "src/base/platform/memory.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/assembler.h"
 #include "src/codegen/macro-assembler.h"
 #include "src/codegen/ppc/constants-ppc.h"
diff --git a/src/execution/ppc/simulator-ppc.h b/src/execution/ppc/simulator-ppc.h
index e3dbd1efa99..bfabd9af5c8 100644
--- a/src/execution/ppc/simulator-ppc.h
+++ b/src/execution/ppc/simulator-ppc.h
@@ -21,7 +21,6 @@
 #include "src/base/hashmap.h"
 #include "src/base/lazy-instance.h"
 #include "src/base/platform/mutex.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/assembler.h"
 #include "src/codegen/ppc/constants-ppc.h"
 #include "src/execution/simulator-base.h"
diff --git a/src/execution/s390/simulator-s390.cc b/src/execution/s390/simulator-s390.cc
index 1e8712d323c..71d63b1b922 100644
--- a/src/execution/s390/simulator-s390.cc
+++ b/src/execution/s390/simulator-s390.cc
@@ -14,8 +14,8 @@
 
 #include "src/base/bits.h"
 #include "src/base/once.h"
+#include "src/base/platform/memory.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/assembler.h"
 #include "src/codegen/macro-assembler.h"
 #include "src/codegen/register-configuration.h"
diff --git a/src/heap/heap.cc b/src/heap/heap.cc
index d8cb75dc5e1..a2b4a7cb789 100644
--- a/src/heap/heap.cc
+++ b/src/heap/heap.cc
@@ -18,8 +18,8 @@
 #include "src/base/logging.h"
 #include "src/base/macros.h"
 #include "src/base/once.h"
+#include "src/base/platform/memory.h"
 #include "src/base/platform/mutex.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/utils/random-number-generator.h"
 #include "src/builtins/accessors.h"
 #include "src/codegen/assembler-inl.h"
diff --git a/src/heap/memory-chunk.cc b/src/heap/memory-chunk.cc
index 200014487e6..0a5ffa17b94 100644
--- a/src/heap/memory-chunk.cc
+++ b/src/heap/memory-chunk.cc
@@ -6,7 +6,6 @@
 
 #include "src/base/logging.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/common/globals.h"
 #include "src/heap/basic-memory-chunk.h"
 #include "src/heap/code-object-registry.h"
diff --git a/src/libplatform/tracing/trace-object.cc b/src/libplatform/tracing/trace-object.cc
index 0aec82673ce..f616679c89e 100644
--- a/src/libplatform/tracing/trace-object.cc
+++ b/src/libplatform/tracing/trace-object.cc
@@ -7,7 +7,6 @@
 #include "include/v8-platform.h"
 #include "src/base/platform/platform.h"
 #include "src/base/platform/time.h"
-#include "src/base/platform/wrappers.h"
 
 namespace v8 {
 namespace platform {
diff --git a/src/numbers/conversions-inl.h b/src/numbers/conversions-inl.h
index fa3fede0fd4..f73d969b324 100644
--- a/src/numbers/conversions-inl.h
+++ b/src/numbers/conversions-inl.h
@@ -17,7 +17,6 @@
 #include "src/base/bits.h"
 #include "src/base/numbers/double.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/numbers/conversions.h"
 #include "src/objects/heap-number-inl.h"
 #include "src/objects/objects-inl.h"
diff --git a/src/numbers/conversions.cc b/src/numbers/conversions.cc
index 433fe61b9fc..902fb53bdcd 100644
--- a/src/numbers/conversions.cc
+++ b/src/numbers/conversions.cc
@@ -11,7 +11,6 @@
 
 #include "src/base/numbers/dtoa.h"
 #include "src/base/numbers/strtod.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/small-vector.h"
 #include "src/bigint/bigint.h"
 #include "src/common/assert-scope.h"
diff --git a/src/objects/backing-store.cc b/src/objects/backing-store.cc
index fd955ed59d0..7a9baa0f366 100644
--- a/src/objects/backing-store.cc
+++ b/src/objects/backing-store.cc
@@ -6,7 +6,6 @@
 
 #include <cstring>
 
-#include "src/base/platform/wrappers.h"
 #include "src/execution/isolate.h"
 #include "src/handles/global-handles.h"
 #include "src/logging/counters.h"
diff --git a/src/objects/bigint.cc b/src/objects/bigint.cc
index 57de312587a..6a8bdc94d54 100644
--- a/src/objects/bigint.cc
+++ b/src/objects/bigint.cc
@@ -146,7 +146,6 @@ class MutableBigInt : public FreshlyAllocatedBigInt {
 OBJECT_CONSTRUCTORS_IMPL(MutableBigInt, FreshlyAllocatedBigInt)
 NEVER_READ_ONLY_SPACE_IMPL(MutableBigInt)
 
-#include "src/base/platform/wrappers.h"
 #include "src/objects/object-macros-undef.h"
 
 bigint::Digits GetDigits(BigIntBase bigint) {
diff --git a/src/objects/fixed-array-inl.h b/src/objects/fixed-array-inl.h
index b97d06aa676..26ffe049419 100644
--- a/src/objects/fixed-array-inl.h
+++ b/src/objects/fixed-array-inl.h
@@ -757,7 +757,6 @@ void TemplateList::set(int index, Object value) {
 }  // namespace internal
 }  // namespace v8
 
-#include "src/base/platform/wrappers.h"
 #include "src/objects/object-macros-undef.h"
 
 #endif  // V8_OBJECTS_FIXED_ARRAY_INL_H_
diff --git a/src/objects/js-array-buffer.cc b/src/objects/js-array-buffer.cc
index c5b45087f0a..3622a30f5d9 100644
--- a/src/objects/js-array-buffer.cc
+++ b/src/objects/js-array-buffer.cc
@@ -4,7 +4,6 @@
 
 #include "src/objects/js-array-buffer.h"
 
-#include "src/base/platform/wrappers.h"
 #include "src/execution/protectors-inl.h"
 #include "src/logging/counters.h"
 #include "src/objects/js-array-buffer-inl.h"
diff --git a/src/objects/shared-function-info-inl.h b/src/objects/shared-function-info-inl.h
index f9ba9084097..43ef0d3d173 100644
--- a/src/objects/shared-function-info-inl.h
+++ b/src/objects/shared-function-info-inl.h
@@ -962,7 +962,6 @@ bool SharedFunctionInfo::are_properties_final() const {
 }  // namespace internal
 }  // namespace v8
 
-#include "src/base/platform/wrappers.h"
 #include "src/objects/object-macros-undef.h"
 
 #endif  // V8_OBJECTS_SHARED_FUNCTION_INFO_INL_H_
diff --git a/src/objects/value-serializer.cc b/src/objects/value-serializer.cc
index 7ec844115da..ba74cbd54af 100644
--- a/src/objects/value-serializer.cc
+++ b/src/objects/value-serializer.cc
@@ -12,7 +12,7 @@
 #include "include/v8-wasm.h"
 #include "src/api/api-inl.h"
 #include "src/base/logging.h"
-#include "src/base/platform/wrappers.h"
+#include "src/base/platform/memory.h"
 #include "src/execution/isolate.h"
 #include "src/flags/flags.h"
 #include "src/handles/global-handles-inl.h"
diff --git a/src/parsing/preparse-data.cc b/src/parsing/preparse-data.cc
index 2ecf7f7979e..7ebdb4e925a 100644
--- a/src/parsing/preparse-data.cc
+++ b/src/parsing/preparse-data.cc
@@ -9,7 +9,6 @@
 #include "src/ast/scopes.h"
 #include "src/ast/variables.h"
 #include "src/base/logging.h"
-#include "src/base/platform/wrappers.h"
 #include "src/handles/handles.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/shared-function-info.h"
diff --git a/src/parsing/scanner.cc b/src/parsing/scanner.cc
index 81894694afb..f6d7bcdee32 100644
--- a/src/parsing/scanner.cc
+++ b/src/parsing/scanner.cc
@@ -11,7 +11,6 @@
 #include <cmath>
 
 #include "src/ast/ast-value-factory.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/strings.h"
 #include "src/numbers/conversions-inl.h"
 #include "src/objects/bigint.h"
diff --git a/src/snapshot/embedded/embedded-file-writer.h b/src/snapshot/embedded/embedded-file-writer.h
index 32f79ff43cd..c124853b6fe 100644
--- a/src/snapshot/embedded/embedded-file-writer.h
+++ b/src/snapshot/embedded/embedded-file-writer.h
@@ -10,6 +10,7 @@
 #include <cstring>
 #include <memory>
 
+#include "src/base/platform/wrappers.h"
 #include "src/base/strings.h"
 #include "src/common/globals.h"
 #include "src/snapshot/embedded/embedded-data.h"
@@ -17,7 +18,6 @@
 #include "src/snapshot/embedded/platform-embedded-file-writer-base.h"
 
 #if defined(V8_OS_WIN64)
-#include "src/base/platform/wrappers.h"
 #include "src/diagnostics/unwinding-info-win64.h"
 #endif  // V8_OS_WIN64
 
diff --git a/src/tracing/trace-event.h b/src/tracing/trace-event.h
index d89d5492771..ed53d55a88b 100644
--- a/src/tracing/trace-event.h
+++ b/src/tracing/trace-event.h
@@ -21,7 +21,6 @@
 #include "include/v8-platform.h"
 #include "src/base/atomicops.h"
 #include "src/base/macros.h"
-#include "src/base/platform/wrappers.h"
 
 // This header file defines implementation details of how the trace macros in
 // trace_event_common.h collect and store trace events. Anything not
diff --git a/src/utils/allocation.cc b/src/utils/allocation.cc
index 370dd50f281..5876de3cb47 100644
--- a/src/utils/allocation.cc
+++ b/src/utils/allocation.cc
@@ -12,7 +12,6 @@
 #include "src/base/logging.h"
 #include "src/base/page-allocator.h"
 #include "src/base/platform/memory.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/sanitizer/lsan-page-allocator.h"
 #include "src/base/sanitizer/lsan-virtual-address-space.h"
 #include "src/base/virtual-address-space.h"
diff --git a/src/utils/memcopy.h b/src/utils/memcopy.h
index ae4701e7354..bc14556558f 100644
--- a/src/utils/memcopy.h
+++ b/src/utils/memcopy.h
@@ -13,7 +13,6 @@
 
 #include "src/base/logging.h"
 #include "src/base/macros.h"
-#include "src/base/platform/wrappers.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/wasm/baseline/arm/liftoff-assembler-arm.h b/src/wasm/baseline/arm/liftoff-assembler-arm.h
index 77a9a45ba1a..9474c65c4f5 100644
--- a/src/wasm/baseline/arm/liftoff-assembler-arm.h
+++ b/src/wasm/baseline/arm/liftoff-assembler-arm.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_ARM_LIFTOFF_ASSEMBLER_ARM_H_
 #define V8_WASM_BASELINE_ARM_LIFTOFF_ASSEMBLER_ARM_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/base/v8-fallthrough.h"
 #include "src/codegen/arm/register-arm.h"
 #include "src/common/globals.h"
diff --git a/src/wasm/baseline/arm64/liftoff-assembler-arm64.h b/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
index 75e15737387..e0fef9b36de 100644
--- a/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
+++ b/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_ARM64_LIFTOFF_ASSEMBLER_ARM64_H_
 #define V8_WASM_BASELINE_ARM64_LIFTOFF_ASSEMBLER_ARM64_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/base/v8-fallthrough.h"
 #include "src/heap/memory-chunk.h"
 #include "src/wasm/baseline/liftoff-assembler.h"
diff --git a/src/wasm/baseline/ia32/liftoff-assembler-ia32.h b/src/wasm/baseline/ia32/liftoff-assembler-ia32.h
index 83b8ded59f5..f82fdda07af 100644
--- a/src/wasm/baseline/ia32/liftoff-assembler-ia32.h
+++ b/src/wasm/baseline/ia32/liftoff-assembler-ia32.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_IA32_LIFTOFF_ASSEMBLER_IA32_H_
 #define V8_WASM_BASELINE_IA32_LIFTOFF_ASSEMBLER_IA32_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/base/v8-fallthrough.h"
 #include "src/codegen/assembler.h"
 #include "src/heap/memory-chunk.h"
diff --git a/src/wasm/baseline/liftoff-assembler.cc b/src/wasm/baseline/liftoff-assembler.cc
index 924ac6e8f45..5d88d60039d 100644
--- a/src/wasm/baseline/liftoff-assembler.cc
+++ b/src/wasm/baseline/liftoff-assembler.cc
@@ -7,7 +7,7 @@
 #include <sstream>
 
 #include "src/base/optional.h"
-#include "src/base/platform/wrappers.h"
+#include "src/base/platform/memory.h"
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/macro-assembler-inl.h"
 #include "src/compiler/linkage.h"
diff --git a/src/wasm/baseline/liftoff-compiler.cc b/src/wasm/baseline/liftoff-compiler.cc
index 8a2c662785a..24ed2824209 100644
--- a/src/wasm/baseline/liftoff-compiler.cc
+++ b/src/wasm/baseline/liftoff-compiler.cc
@@ -6,7 +6,6 @@
 
 #include "src/base/enum-set.h"
 #include "src/base/optional.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/assembler-inl.h"
 // TODO(clemensb): Remove dependences on compiler stuff.
 #include "src/codegen/external-reference.h"
diff --git a/src/wasm/baseline/loong64/liftoff-assembler-loong64.h b/src/wasm/baseline/loong64/liftoff-assembler-loong64.h
index ce590eb9d7a..55a5459912e 100644
--- a/src/wasm/baseline/loong64/liftoff-assembler-loong64.h
+++ b/src/wasm/baseline/loong64/liftoff-assembler-loong64.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_LOONG64_LIFTOFF_ASSEMBLER_LOONG64_H_
 #define V8_WASM_BASELINE_LOONG64_LIFTOFF_ASSEMBLER_LOONG64_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/machine-type.h"
 #include "src/heap/memory-chunk.h"
 #include "src/wasm/baseline/liftoff-assembler.h"
diff --git a/src/wasm/baseline/mips/liftoff-assembler-mips.h b/src/wasm/baseline/mips/liftoff-assembler-mips.h
index dbb18f5c6fc..8bad92d522b 100644
--- a/src/wasm/baseline/mips/liftoff-assembler-mips.h
+++ b/src/wasm/baseline/mips/liftoff-assembler-mips.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_MIPS_LIFTOFF_ASSEMBLER_MIPS_H_
 #define V8_WASM_BASELINE_MIPS_LIFTOFF_ASSEMBLER_MIPS_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/heap/memory-chunk.h"
 #include "src/wasm/baseline/liftoff-assembler.h"
 #include "src/wasm/wasm-objects.h"
diff --git a/src/wasm/baseline/mips64/liftoff-assembler-mips64.h b/src/wasm/baseline/mips64/liftoff-assembler-mips64.h
index 1aaeb0e67f4..b7d47506afb 100644
--- a/src/wasm/baseline/mips64/liftoff-assembler-mips64.h
+++ b/src/wasm/baseline/mips64/liftoff-assembler-mips64.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_MIPS64_LIFTOFF_ASSEMBLER_MIPS64_H_
 #define V8_WASM_BASELINE_MIPS64_LIFTOFF_ASSEMBLER_MIPS64_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/machine-type.h"
 #include "src/heap/memory-chunk.h"
 #include "src/wasm/baseline/liftoff-assembler.h"
diff --git a/src/wasm/baseline/ppc/liftoff-assembler-ppc.h b/src/wasm/baseline/ppc/liftoff-assembler-ppc.h
index abd97a13bdd..cd2684bd719 100644
--- a/src/wasm/baseline/ppc/liftoff-assembler-ppc.h
+++ b/src/wasm/baseline/ppc/liftoff-assembler-ppc.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_PPC_LIFTOFF_ASSEMBLER_PPC_H_
 #define V8_WASM_BASELINE_PPC_LIFTOFF_ASSEMBLER_PPC_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/base/v8-fallthrough.h"
 #include "src/codegen/assembler.h"
 #include "src/heap/memory-chunk.h"
diff --git a/src/wasm/baseline/riscv/liftoff-assembler-riscv.h b/src/wasm/baseline/riscv/liftoff-assembler-riscv.h
index bfb90abb5e9..925b02a4624 100644
--- a/src/wasm/baseline/riscv/liftoff-assembler-riscv.h
+++ b/src/wasm/baseline/riscv/liftoff-assembler-riscv.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_RISCV_LIFTOFF_ASSEMBLER_RISCV_H_
 #define V8_WASM_BASELINE_RISCV_LIFTOFF_ASSEMBLER_RISCV_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/heap/memory-chunk.h"
 #include "src/wasm/baseline/liftoff-assembler.h"
 #include "src/wasm/wasm-objects.h"
diff --git a/src/wasm/baseline/riscv/liftoff-assembler-riscv32.h b/src/wasm/baseline/riscv/liftoff-assembler-riscv32.h
index c0aa014f5a9..28803b9f101 100644
--- a/src/wasm/baseline/riscv/liftoff-assembler-riscv32.h
+++ b/src/wasm/baseline/riscv/liftoff-assembler-riscv32.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_RISCV_LIFTOFF_ASSEMBLER_RISCV32_H_
 #define V8_WASM_BASELINE_RISCV_LIFTOFF_ASSEMBLER_RISCV32_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/heap/memory-chunk.h"
 #include "src/wasm/baseline/liftoff-assembler.h"
 #include "src/wasm/baseline/riscv/liftoff-assembler-riscv.h"
diff --git a/src/wasm/baseline/riscv/liftoff-assembler-riscv64.h b/src/wasm/baseline/riscv/liftoff-assembler-riscv64.h
index b9d18c127cc..738c3a8126e 100644
--- a/src/wasm/baseline/riscv/liftoff-assembler-riscv64.h
+++ b/src/wasm/baseline/riscv/liftoff-assembler-riscv64.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_RISCV_LIFTOFF_ASSEMBLER_RISCV64_H_
 #define V8_WASM_BASELINE_RISCV_LIFTOFF_ASSEMBLER_RISCV64_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/heap/memory-chunk.h"
 #include "src/wasm/baseline/liftoff-assembler.h"
 #include "src/wasm/baseline/riscv/liftoff-assembler-riscv.h"
diff --git a/src/wasm/baseline/s390/liftoff-assembler-s390.h b/src/wasm/baseline/s390/liftoff-assembler-s390.h
index 20c712fd6fb..f4e06ad1029 100644
--- a/src/wasm/baseline/s390/liftoff-assembler-s390.h
+++ b/src/wasm/baseline/s390/liftoff-assembler-s390.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_S390_LIFTOFF_ASSEMBLER_S390_H_
 #define V8_WASM_BASELINE_S390_LIFTOFF_ASSEMBLER_S390_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/base/v8-fallthrough.h"
 #include "src/codegen/assembler.h"
 #include "src/heap/memory-chunk.h"
diff --git a/src/wasm/baseline/x64/liftoff-assembler-x64.h b/src/wasm/baseline/x64/liftoff-assembler-x64.h
index 544ec670b96..0af5753584a 100644
--- a/src/wasm/baseline/x64/liftoff-assembler-x64.h
+++ b/src/wasm/baseline/x64/liftoff-assembler-x64.h
@@ -5,7 +5,6 @@
 #ifndef V8_WASM_BASELINE_X64_LIFTOFF_ASSEMBLER_X64_H_
 #define V8_WASM_BASELINE_X64_LIFTOFF_ASSEMBLER_X64_H_
 
-#include "src/base/platform/wrappers.h"
 #include "src/base/v8-fallthrough.h"
 #include "src/codegen/assembler.h"
 #include "src/codegen/cpu-features.h"
diff --git a/src/wasm/c-api.cc b/src/wasm/c-api.cc
index b573f4c87c5..0ea84e39ac2 100644
--- a/src/wasm/c-api.cc
+++ b/src/wasm/c-api.cc
@@ -28,7 +28,6 @@
 #include "include/libplatform/libplatform.h"
 #include "include/v8-initialization.h"
 #include "src/api/api-inl.h"
-#include "src/base/platform/wrappers.h"
 #include "src/builtins/builtins.h"
 #include "src/compiler/wasm-compiler.h"
 #include "src/objects/call-site-info-inl.h"
diff --git a/src/wasm/module-decoder-impl.h b/src/wasm/module-decoder-impl.h
index 625481bd403..227fc327c2b 100644
--- a/src/wasm/module-decoder-impl.h
+++ b/src/wasm/module-decoder-impl.h
@@ -9,6 +9,7 @@
 #ifndef V8_WASM_MODULE_DECODER_IMPL_H_
 #define V8_WASM_MODULE_DECODER_IMPL_H_
 
+#include "src/base/platform/wrappers.h"
 #include "src/logging/counters.h"
 #include "src/strings/unicode.h"
 #include "src/utils/ostreams.h"
diff --git a/src/wasm/wasm-code-manager.cc b/src/wasm/wasm-code-manager.cc
index 35a9a23dab0..51874973e1d 100644
--- a/src/wasm/wasm-code-manager.cc
+++ b/src/wasm/wasm-code-manager.cc
@@ -13,6 +13,7 @@
 #include "src/base/iterator.h"
 #include "src/base/macros.h"
 #include "src/base/platform/platform.h"
+#include "src/base/platform/wrappers.h"
 #include "src/base/small-vector.h"
 #include "src/base/string-format.h"
 #include "src/base/vector.h"
@@ -42,7 +43,6 @@
 #include "src/wasm/wasm-objects.h"
 
 #if defined(V8_OS_WIN64)
-#include "src/base/platform/wrappers.h"
 #include "src/diagnostics/unwinding-info-win64.h"
 #endif  // V8_OS_WIN64
 
diff --git a/src/wasm/wasm-engine.cc b/src/wasm/wasm-engine.cc
index 3ba232caf5f..ddb45653511 100644
--- a/src/wasm/wasm-engine.cc
+++ b/src/wasm/wasm-engine.cc
@@ -30,7 +30,6 @@
 #include "src/wasm/wasm-objects-inl.h"
 
 #ifdef V8_ENABLE_WASM_GDB_REMOTE_DEBUGGING
-#include "src/base/platform/wrappers.h"
 #include "src/debug/wasm/gdb-server/gdb-server.h"
 #endif  // V8_ENABLE_WASM_GDB_REMOTE_DEBUGGING
 
diff --git a/src/web-snapshot/web-snapshot.cc b/src/web-snapshot/web-snapshot.cc
index 62403915b8d..f2f91ee47cd 100644
--- a/src/web-snapshot/web-snapshot.cc
+++ b/src/web-snapshot/web-snapshot.cc
@@ -12,7 +12,6 @@
 #include "include/v8-primitive.h"
 #include "include/v8-script.h"
 #include "src/api/api-inl.h"
-#include "src/base/platform/wrappers.h"
 #include "src/handles/handles.h"
 #include "src/logging/runtime-call-stats-scope.h"
 #include "src/objects/bigint.h"
diff --git a/src/zone/accounting-allocator.cc b/src/zone/accounting-allocator.cc
index cdfe0304b94..d87981d632e 100644
--- a/src/zone/accounting-allocator.cc
+++ b/src/zone/accounting-allocator.cc
@@ -9,7 +9,6 @@
 #include "src/base/bounded-page-allocator.h"
 #include "src/base/logging.h"
 #include "src/base/macros.h"
-#include "src/base/platform/wrappers.h"
 #include "src/utils/allocation.h"
 #include "src/zone/zone-compression.h"
 #include "src/zone/zone-segment.h"
diff --git a/src/zone/zone-list-inl.h b/src/zone/zone-list-inl.h
index fba12abd174..7cf9bec1aef 100644
--- a/src/zone/zone-list-inl.h
+++ b/src/zone/zone-list-inl.h
@@ -7,7 +7,6 @@
 
 #include "src/base/macros.h"
 #include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/utils/memcopy.h"
 #include "src/zone/zone-list.h"
 
-- 
2.35.1

